astropy/coordinates/representation.py

```diff
@@ -153,29 +153,44 @@ def _unitstr(self):
         return unitstr
 
     def __str__(self):
-        if self.isscalar and len(set(self._units.values())) > 1:
-            return '({0})'.format(', '.join(
-                ['{0}'.format(getattr(self, component))
-                 for component in self.components]))
-        else:
-            return '{0} {1:s}'.format(self._values, self._unitstr)
+        return '{0} {1:s}'.format(self._values, self._unitstr)
 
     def __repr__(self):
-        if self.isscalar:
-            return '<{0} {1}>'.format(
-                self.__class__.__name__,
-                ', '.join(['{0}={1}'.format(component,
-                                            getattr(self, component))
-                           for component in self.components]))
+        prefixstr = '    '
 
+        if self._values.shape == ():
+            v = [tuple([self._values[nm] for nm in self._values.dtype.names])]
+            v = np.array(v, dtype=self._values.dtype)
         else:
-            prefixstr = '    '
-            arrstr = np.array2string(self._values, separator=', ',
-                                     prefix=prefixstr)
-
-            return '<{0} ({1}) in {2:s}\n{3}{4}>'.format(
-                self.__class__.__name__, ', '.join(self.components),
-                self._unitstr, prefixstr, arrstr)
+            v = self._values
+
+        fstyle = lambda x: str(x)
+        names = self._values.dtype.names
+        if len(names) == 2:
+            fmt = {'numpystr' : lambda x: "(" + \
+                                            np.array2string(x[names[0]], style=fstyle) + \
+                                            ', ' + \
+                                            np.array2string(x[names[1]], style=fstyle) + \
+                                            ")"}
+        else:
+            fmt = {'numpystr' : lambda x: "(" + \
+                                            np.array2string(x[names[0]], style=fstyle) + \
+                                            ', ' + \
+                                            np.array2string(x[names[1]], style=fstyle) + \
+                                            ', ' + \
+                                            np.array2string(x[names[2]], style=fstyle) + \
+                                            ")"}
+
+        arrstr = np.array2string(v, formatter=fmt,
+                                 separator=', ',
+                                 prefix=prefixstr)
+
+        if self._values.shape == ():
+            arrstr = arrstr[1:-1]
+
+        return '<{0} ({1}) in {2:s}\n{3}{4}>'.format(
```

Comment:
I guess before this line I would do either one of the following:
``
unitstr = "in " + self._unitstr if self._unitstr is not None else ""
```
or
```
unitstr = 'dimensionles' if self._unitstr is None else self._unitstr
```

Of course, in the first example, the format string should be adjusted appropriately.